examples/platforms/nrf528xx/CMakeLists.txt

```diff
@@ -0,0 +1,132 @@
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/usb/config
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/usb/class/cdc
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/usb/class/cdc/acm
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/utf_converter
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/nrfx
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/nrfx/hal
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/nrfx/drivers
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/nrfx/drivers/include
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/nrfx/mdk
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/nrfx/soc
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/softdevice/s140/headers
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/softdevice/s140/headers/nrf52
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/drivers/radio
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/drivers/radio/fem/three_pin_gpio
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/drivers/radio/hal
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/drivers/radio/mac_features
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/drivers/radio/mac_features/ack_generator
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/drivers/radio/rsch
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/drivers/radio/rsch/raal
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/drivers/radio/rsch/raal/softdevice
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/drivers/radio/platform/lp_timer
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/drivers/radio/platform/temperature
+)
+
+set(NRF_NONMBEDTLS_LIBS
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/lib/libmbedcrypto_glue.a
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/lib/libmbedcrypto_glue_cc310.a
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/lib/libmbedcrypto_glue_vanilla.a
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/lib/libmbedcrypto_vanilla_backend.a
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/lib/libmbedtls_tls_vanilla.a
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/lib/libmbedtls_x509_vanilla.a
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/lib/libmbedtls_base_vanilla.a
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/lib/libnrf_cc310_platform_0.9.2.a
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/lib/libmbedcrypto_cc310_backend.a
+    ${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/lib/libmbedtls_base_vanilla.a
+)
```

Comment:
We need to avoid passing dependencies through variables, especially when it is not under `$CMAKE_CURRENT_SOURCE_DIR`.
Create an interface target in `third_party/NordicSemiconductor/` and pass the include directories through `target_include_directories` and `target_link_libraries`